server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServerImpl.java

```diff
@@ -65,11 +73,21 @@ public GuiceJamesServerImpl overrideWith(Module... overrides) {
     @Override
     public void start() throws Exception {
         Injector injector = Guice.createInjector(module);
+        logInjectedElementsWithNoScope(injector);
+
         preDestroy = injector.getInstance(Key.get(new TypeLiteral<Stager<PreDestroy>>() {}));
         injector.getInstance(ConfigurationsPerformer.class).initModules();
         guiceProbeProvider = injector.getInstance(GuiceProbeProvider.class);
     }
 
+    private void logInjectedElementsWithNoScope(Injector injector) {
+        injector.getAllBindings().entrySet().stream()
+            .filter(entry -> entry.getValue() instanceof ConstructorBinding)
```

Comment:
I don't agree on general case, espacially if it leads to something more readable